dds_cli/__main__.py

```diff
@@ -372,356 +264,243 @@ def ls(click_ctx, project, folder, projects, size, username, usage, sort, tree,
 @click.pass_obj
-def login(click_ctx, username):
-    """Renew the authentication token stored in the '.dds_cli_token' file.
-
-    Run this command before running the cli in a non interactive fashion as this enables the longest
-    possible session time before a password needs to be entered again.
-    """
-    no_prompt = click_ctx.get("NO_PROMPT", False)
-    if no_prompt:
-        LOG.warning("The --no-prompt flag is ignored for `dds auth login`")
-    try:
-        with dds_cli.auth.Auth(username=username):
-            # Authentication token renewed in the init method.
-            LOG.info("[green] :white_check_mark: Authentication token renewed![/green]")
-    except (
-        dds_cli.exceptions.APIError,
-        dds_cli.exceptions.AuthenticationError,
-        dds_cli.exceptions.DDSCLIException,
-    ) as err:
-        LOG.error(err)
-        sys.exit(1)
-
-
-@auth.command()
-def logout():
-    """Remove the saved authentication token by deleting the '.dds_cli_token' file."""
-    try:
-        with dds_cli.auth.Auth(username=None, authenticate=False) as authenticator:
-            authenticator.logout()
-            LOG.info("[green]Successfully logged out![/green]")
-    except dds_cli.exceptions.DDSCLIException as err:
-        LOG.error(err)
-        sys.exit(1)
-
-
-@auth.command()
-def info():
-    """Print info on saved authentication token validity and age."""
-    try:
-        with dds_cli.auth.Auth(username=None, authenticate=False) as authenticator:
-            authenticator.check()
-    except dds_cli.exceptions.DDSCLIException as err:
-        LOG.error(err)
-        sys.exit(1)
+def project_group_command(_):
+    """Group command: dds project. Manage projects."""
 
 
-# COMMAND##########################################################################
-#### PROJECT ############################################################ PROJECT #
-###################################################################################
-@dds_main.group(invoke_without_command=True)
-@click.pass_obj
-def project(click_ctx):
-    """Manage projects"""
-    pass
+# ************************************************************************************************ #
+# PROJECT COMMANDS ************************************************************** PROJECT COMMANDS #
+# ************************************************************************************************ #
 
 
-def common_options_project(f):
-    options = [
-        click.option("--project", "-p", required=True, type=str, help="Project ID."),
-    ]
-    return functools.reduce(lambda x, opt: opt(x), options, f)
+# -- dds project ls -- #
+@project_group_command.command(name="ls")
+# Options
+@username_option()
+@sort_projects_option()
+# Flags
+@usage_flag(help_message="Show the usage for available projects, in GBHours and cost. ")
+@json_flag(help_message="Output project list as json.")  # users, json, tree
+@click.pass_context
```

Comment:
I think, this is done on purpose because it is the sole non-group command on the top-level, but just to double-check: You deliberately used `click.pass_context` here instead of `click.pass_obj`?

Reply:
Correct. `click.pass_obj` passes a dict, and `click.pass_context` passes an object that can then perform the `invoke` of another command.